lib/rucio/tests/test_reaper2.py

```diff
@@ -203,3 +209,130 @@ def test_reaper_multi_vo(vo):
     reaper(once=True, rses=[], include_rses=both_rses, exclude_rses=None)
     assert len(list(replica_core.list_replicas(dids=dids1, rse_expression=both_rses))) == 200
     assert len(list(replica_core.list_replicas(dids=dids2, rse_expression=both_rses))) == 200
+
+
+def test_archive_removal_impact_on_constituents(rse_factory, file_factory, mock_scope, root_account):
+    rse_name, rse_id = rse_factory.make_mock_rse()
+    scope = mock_scope
+    account = root_account
+
+    # Create an 2 archives and 4 files:
+    # - One only exists in the first archive
+    # - One in both, plus another replica, which is not in an archive
+    # - One in both, plus another replica, which is not in an archive; and this replica has expired
+    # - One in both, plus another replica, which is not in an archive; and this replica has expired; but a replication rule exists on this second replica
```

Comment:
This condition shouldn't occur if the rule already exists

Reply:
Could you please elaborate ?

Reply:
I think what Cedric means is that the forth one 
```
One in both, plus another replica, which is not in an archive; and this replica has expired; but a replication rule exists on this second replica
```
cannot exist, since there cannot be an expired replica covered by a replication rule.

Reply:
Should I remove this case from the test then ? it ensures that replicas protected by rules are not removed by reaper. But it's better to remove it completely if such case cannot arrive.

Reply:
In theory such a case cannot arrive, but in pricniple it makes sense that the reaper protects for it as well, just in case some other workflow is not acting as expected. I would leave it. @cserf what do you think?